src/track/beatmap.cpp

```diff
@@ -268,6 +272,89 @@ double BeatMap::findNthBeat(double dSamples, int n) const {
     return -1;
 }
 
+QPair<double, double> BeatMap::findPrevNextBeats(double dSamples) const {
```

Comment:
You should not use QPair hier, since it hides the content of the structure members. 
Since the QPair is only used as return value I think it is most effective to use a prototype like this: 

void BeatMap::findPrevNextBeats(double dSamples, double* pPrevSamples, double* pNextSamples ) const;